### ISSUE TITLE
add ability to save coil, liquid and flavors

### ISSUE THREAD

**Issue by stasbar** [vape-tool/VapeTool-Webapp#36]
Each logged-in user can save coils he creates into FirebaseDatabase (same as photos/posts/links). There should be "Save" button in Coil Calculator and Liquid-Bledner pages.
Fortunately `src/models/coil.ts` already store the state in Coil class, same in `src/models/liquid.ts`.

Coils are very similar to Photos, because they can also contain photo of the coil. So the process of saving a coil is the same as [uploading a photo](https://github.com/vape-tool/VapeTool-Webapp/blob/master/src/services/items.ts#L201). In the SaveCoil modal, there should be the place to upload a coil photo (just reuse as much as possible from Uploading Photo).

Liquids are simpler since they don't contain photos. They are similar to uploading [Post or Link](https://github.com/vape-tool/VapeTool-Webapp/blob/master/src/services/items.ts#L138)

Just remember to save types defined in `@vapetool/types`, so FirebaseCoil, FirebaseLiquid and FirebaseFlavor because they are the actual types saved in Firebase Database, and might differ slightly from App-Domain models defined in `@/types`.

Saved in Database
```
import {
  Photo as FirebasePhoto,
  Post as FirebasePost,
  Link as FirebaseLink,
  Coil as FirebaseCoil,
  Liquid as FirebaseLiquid,
  Flavor as FirebaseFlavor,
} from '@vapetool/types';
```

Within app domain, we are opearating on:
```
import {
  Photo,
  Post,
  Link,
  Coil,
  Liquid,
  Flavor,
} from '@/types';
```

**Comment by stasbar** [comment]
Before each save, there should be calculation performed, when someone change the wraps number, and forget to press calculate it should be illegal to save such inconsistent coil state.

**Comment by gbaranski** [comment]
closed by #69
